lib/emscripten/src/syscalls/unix.rs

```diff
@@ -452,13 +445,15 @@ pub fn ___syscall54(ctx: &mut Ctx, _which: c_int, mut varargs: VarArgs) -> c_int
                 ret
             }
         }
-        _ => {
-            debug!(
-                "emscripten::___syscall54 -> non implemented case {}",
-                request
-            );
+        TIOCSPGRP => {
+            debug!("ret(TIOCSPGRP): (noop)");
             0
         }
+        _ => {
+            let ret = unsafe { ioctl(fd, request as _, argp_ptr) };
```

Comment:
We can't do this (as exposed in the previous comment).

Reply:
If you want to do it in a generic way, we need a proper mapping of fixed values to the `libc` values.

Reply:
What targets are we focused or are we going to be focused in the short term at this moment beyond x86_64? AArch64? Maybe ARM in 32 bits? What flavour (v6 for Rapsberry Pi 1? v7? Cortex_A42?)

Reply:
x86_64 and AArch64 are the main ones... why is the target relevant?

Reply:
To know to what platforms should I do the mapping.

Reply:
I think you don't really need to know the platform, you can use libc values for the real calls.

Here's an example on how:

```rust
use libc::TIOCSPGRP;

WASM_TIOCSPGRP = 21523;

// This is not syntactically valid:
let MAPPING = {
   WASM_TIOCSPGRP: TIOCSPGRP
}

// ...

let ret = unsafe { ioctl(fd, MAPPING[request], argp_ptr) };

```

Reply:
Touchée, I got distracted with the previous comment about the values can be different for each platform. Yes, `libc` can do the mapping by ourselves.

I need to left work for today and go to the LGTB Proud parade to distract myself a bit :sweat_smile: :unicorn: :rainbow_flag: :-)

Reply:
Done :-)